bench/xs_bench.ml

```diff
@@ -177,76 +197,96 @@ let private_path = prefix ^ "/xapi"
+			frontend_path_of_device x client
+			>>= fun fe ->
+			backend_path_of_device x client
+			>>= fun be ->
+			backend_error_path_of_device x client
+			>>= fun ber ->
+			frontend_error_path_of_device x client
+			>>= fun fer ->
 			Lwt_list.iter_s (rm xs) [ fe; be; ber; Filename.dirname fer ]
 		)
 
 let hard_shutdown device client =
-	lwt () = hard_shutdown_request device client in
-	lwt () = rm_device_state device client in
-	return ()
+	hard_shutdown_request device client
+	>>= fun () ->
+	rm_device_state device client
 
 let add device client =
 	let backend_list = []
 	and frontend_list = []
 	and private_list = [] in
 
-	lwt frontend_path = frontend_path_of_device device client in
-	lwt backend_path = backend_path_of_device device client in
+	frontend_path_of_device device client
+	>>= fun frontend_path ->
+	backend_path_of_device device client
+	>>= fun backend_path ->
 	let hotplug_path = get_hotplug_path device in
 	let private_data_path = get_private_data_path_of_device device in
-	lwt () = transaction client
+	transaction client
 		(fun xs ->
-			lwt _ = exists (Printf.sprintf "/local/domain/%d/vm" device.backend.domid) xs in
-			lwt _ = exists frontend_path xs in
-			lwt () = try_lwt lwt () = rm xs frontend_path in return () with _ -> return () in
-			lwt () = try_lwt lwt () = rm xs backend_path in return () with _ -> return () in
+			exists (Printf.sprintf "/local/domain/%d/vm" device.backend.domid) xs
+			>>= fun _ ->
+			exists frontend_path xs
+			>>= fun _ ->
+			Lwt.catch (fun () -> rm xs frontend_path) (fun _ -> return ())
+			>>= fun () ->
+			Lwt.catch (fun () -> rm xs backend_path) (fun _ -> return ())
+			>>= fun () ->
 
 			(* CA-16259: don't clear the 'hotplug_path' because this is where we
 			   record our own use of /dev/loop devices. Clearing this causes us to leak
 			   one per PV .iso *)
-			lwt () = mkdir xs frontend_path in
-			lwt () = setperms xs frontend_path (Xs_protocol.ACL.({owner = device.frontend.domid; other = NONE; acl = [ device.backend.domid, READ ]})) in
-			lwt () = mkdir xs backend_path in
-			lwt () = setperms xs backend_path (Xs_protocol.ACL.({owner = device.backend.domid; other = NONE; acl = [ device.frontend.domid, READ ]})) in
-			lwt () = mkdir xs hotplug_path in
-			lwt () = setperms xs hotplug_path (Xs_protocol.ACL.({owner = device.backend.domid; other = NONE; acl = []})) in
-			lwt () = Lwt_list.iter_s (fun (x, y) -> write xs (frontend_path ^ "/" ^ x) y)
-		        (("backend", backend_path) :: frontend_list) in
-			lwt () = Lwt_list.iter_s (fun (x, y) -> write xs (backend_path ^ "/" ^ x) y)
-				(("frontend", frontend_path) :: backend_list) in
-			lwt () = mkdir xs private_data_path in
-			lwt () = setperms xs private_data_path (Xs_protocol.ACL.({owner = device.backend.domid; other = NONE; acl = []})) in
-			lwt () = Lwt_list.iter_s (fun (x, y) -> write xs (private_data_path ^ "/" ^ x) y)
+			mkdir xs frontend_path
+			>>= fun () ->
```

Comment:
Personally, when we ignore the result, I like to throw the boilerplate at the end of the previous expression.
I am not necessarily suggesting to do it here, I only believe that it makes the code less cluttered and more immediately sequentially readable.